main.py

```diff
@@ -2,61 +2,98 @@
-    #     script = service.fetch_script(f'{BASE_URL}?getconfigupdate&equipid=1')
-    #     if script:
-    #         if service.is_valid_bash(script):
-    #             with open('script.sh', 'wb') as f:
-    #                 f.write(script.encode())
-    #             service.run_bash('./script.sh')
-    #             os.remove('./script.sh')
-    #             logger.info('Script executed successfully')
-    #         elif service.is_valid_python(script):
-    #             with open('script.py', 'wb') as f:
-    #                 f.write(script.encode())
-    #             service.run_python('./script.py')
-    #             os.remove('./script.py')
-    #             logger.info('Script executed successfully')
-    # except Exception as e:
-    #     logger.error(e)
 
+logging.basicConfig(filename=f'./assets/logs/{date.today()}.log', level=logging.DEBUG)
+logger = logging.getLogger('main')
+
+logger.setLevel(logging.DEBUG)
+
+def set_brightness():
+    if darkdetect.isDark():
+        sbc.fade_brightness(50)
+    else:
+        sbc.fade_brightness(100, increment = 10)
+    
+
+def delete_old_logs():
+    directory = './assets/logs'
+    for filename in os.scandir(directory):
+        if not filename.is_file():
+            continue
+        date_created = filename.name[:10]
+        week_ago = date.today() - timedelta(days=7)
+        if date_created == str(week_ago):
+            os.remove(os.path.join(directory, filename.name))
+
+# def filter_video_data(video_data):
+#     new_video_data = []
+#     for i in video_data:
+#         if 'locations' not in i:
+#             logger.error(f"Video object {i} has no 'locations' attribute.")
+#             continue
+#         for city in i['locations']:
+#             if 'enname' not in city:
+#                 logger.error(f"City object {city} has no 'enname' attribute.")
+#                 continue
+#             if city['enname'] == current_city:
+#                 new_video_data.append(i)
+#     return new_video_data
+
+
+# def run_maintenance_script():
+#     try:
+#         logger.info('Fetching script...')
+#         script = service.fetch_script(f'{BASE_URL}?getconfigupdate&equipid=1')
+#         if script:
+#             if service.is_valid_bash(script):
+#                 with open('script.sh', 'wb') as f:
+#                     f.write(script.encode())
+#                 service.run_bash('./script.sh')
+#                 os.remove('./script.sh')
+#                 logger.info('Script executed successfully')
+#             elif service.is_valid_python(script):
+#                 with open('script.py', 'wb') as f:
+#                     f.write(script.encode())
+#                 service.run_python('./script.py')
+#                 os.remove('./script.py')
+#                 logger.info('Script executed successfully')
+#     except Exception as e:
+#         logger.error(e)
+
+
+
+def main():
+    # run_maintenance_script()
+    delete_old_logs()
+    set_brightness()
```

Comment:
You need to dynamically update brightness. This screen will be working for the whole day, so you need to set it on timer. That means, it should probably not be in `main`.

Reply:
✅